### PR TITLE
Fxg/ssp

### PR DESCRIPTION
**Pull request by fxgiraldo**
@jkozdon @lcw ,  this pull request runs fast on the GPU and passed all tests on the AWS server.  Let me know what you think. This pull request obviates the need for PR #236

### CODE REVIEW

File: test/DGmethods/runtests.jl [hunk 1]
```diff
@@ -42,7 +44,9 @@ using MPI, Test
                  (3, "Euler/isentropic_vortex_standalone_bc.jl")
                  (3, "conservation/sphere.jl")
                  (3, "compressible_Navier_Stokes/mms_bc.jl")
-                 (2, "sphere/advection_sphere.jl")
+                 (2, "sphere/advection_sphere_lsrk.jl")
+                 (2, "sphere/advection_sphere_ssp32.jl")
```
Comment (jkozdon) [inline test/DGmethods/runtests.jl]:
```suggestion
                 (2, "sphere/advection_sphere_ssp33.jl")
```

File: test/DGmethods/runtests.jl [hunk 2]
```diff
@@ -42,7 +44,9 @@ using MPI, Test
...
+                 (2, "sphere/advection_sphere_ssp43.jl")
```
Comment (jkozdon) [inline test/DGmethods/runtests.jl]:
```suggestion
                 (2, "sphere/advection_sphere_ssp34.jl")
```

File: test/DGmethods/runtests.jl [hunk 3]
```diff
@@ -24,7 +24,9 @@ using MPI, Test
                  (1, "conservation/sphere.jl")
                  (1, "compressible_Navier_Stokes/mms_bc.jl")
                  (3, "compressible_Navier_Stokes/mms_bc.jl")
-                 (1, "sphere/advection_sphere.jl")
+                 (1, "sphere/advection_sphere_lsrk.jl")
+                 (1, "sphere/advection_sphere_ssp32.jl")
```
Comment (jkozdon) [inline test/DGmethods/runtests.jl]:
```suggestion
                 (1, "sphere/advection_sphere_ssp33.jl")
```

File: test/DGmethods/runtests.jl [hunk 4]
```diff
@@ -24,7 +24,9 @@ using MPI, Test
...
+                 (1, "sphere/advection_sphere_ssp43.jl")
```
Comment (jkozdon) [inline test/DGmethods/runtests.jl]:
```suggestion
                 (1, "sphere/advection_sphere_ssp34.jl")
```

**Comment by jkozdon** [comment]
We may want squash your commits, or at least remove the "WIP" in the comment for 8f8417d993136dc6962b8099a4d4bc1892c930e0

File: test/DGmethods/sphere/advection_sphere_lsrk.jl [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+using CLIMA.StrongStabilityPreservingRungeKuttaMethod
+using CLIMA.ODESolvers
+using CLIMA.GenericCallbacks
+using CLIMA.Vtk
+using LinearAlgebra
+using StaticArrays
+using Logging, Printf, Dates
+using Random
+
+@static if Base.find_package("CuArrays") !== nothing
```
Comment (jkozdon) [inline test/DGmethods/sphere/advection_sphere_lsrk.jl]: if #246 lands first this should be
```suggestion
@static if haspkg("CUDAnative")
 ```

File: test/DGmethods/sphere/advection_sphere_lsrk.jl [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+    Sys.iswindows() || (isinteractive() && MPI.finalize_atexit())
+    mpicomm=MPI.COMM_WORLD
+
+    ll = uppercase(get(ENV, "JULIA_LOG_LEVEL", "INFO"))
+    loglevel = ll == "DEBUG" ? Logging.Debug :
+    ll == "WARN"  ? Logging.Warn  :
+    ll == "ERROR" ? Logging.Error : Logging.Info
+    logger_stream = MPI.Comm_rank(mpicomm) == 0 ? stderr : devnull
+    global_logger(ConsoleLogger(logger_stream, loglevel))
+    @static if Base.find_package("CUDAnative") !== nothing
```
Comment (jkozdon) [inline test/DGmethods/sphere/advection_sphere_lsrk.jl]: if #246 lands first this should be
```suggestion
@static if haspkg("CUDAnative")
 ```

File: test/DGmethods/sphere/advection_sphere_ssp33.jl [hunk 7]
```diff
@@ -0,0 +1,384 @@
...
+using CLIMA.StrongStabilityPreservingRungeKuttaMethod
+using CLIMA.ODESolvers
+using CLIMA.GenericCallbacks
+using CLIMA.Vtk
+using LinearAlgebra
+using StaticArrays
+using Logging, Printf, Dates
+using Random
+
+@static if Base.find_package("CuArrays") !== nothing
```
Comment (jkozdon) [inline test/DGmethods/sphere/advection_sphere_ssp33.jl]: if #246 lands first this should be
```suggestion
@static if haspkg("CUDAnative")
 ```

File: test/DGmethods/sphere/advection_sphere_ssp33.jl [hunk 8]
```diff
@@ -0,0 +1,384 @@
...
+    Sys.iswindows() || (isinteractive() && MPI.finalize_atexit())
+    mpicomm=MPI.COMM_WORLD
+
+    ll = uppercase(get(ENV, "JULIA_LOG_LEVEL", "INFO"))
+    loglevel = ll == "DEBUG" ? Logging.Debug :
+    ll == "WARN"  ? Logging.Warn  :
+    ll == "ERROR" ? Logging.Error : Logging.Info
+    logger_stream = MPI.Comm_rank(mpicomm) == 0 ? stderr : devnull
+    global_logger(ConsoleLogger(logger_stream, loglevel))
+    @static if Base.find_package("CUDAnative") !== nothing
```
Comment (jkozdon) [inline test/DGmethods/sphere/advection_sphere_ssp33.jl]: if #246 lands first this should be
```suggestion
@static if haspkg("CUDAnative")
 ```

File: test/DGmethods/sphere/advection_sphere_ssp34.jl [hunk 9]
```diff
@@ -0,0 +1,385 @@
...
+using CLIMA.StrongStabilityPreservingRungeKuttaMethod
+using CLIMA.ODESolvers
+using CLIMA.GenericCallbacks
+using CLIMA.Vtk
+using LinearAlgebra
+using StaticArrays
+using Logging, Printf, Dates
+using Random
+
+@static if Base.find_package("CuArrays") !== nothing
```
Comment (jkozdon) [inline test/DGmethods/sphere/advection_sphere_ssp34.jl]: if #246 lands first this should be
```suggestion
@static if haspkg("CUDAnative")
 ```

File: test/DGmethods/sphere/advection_sphere_ssp34.jl [hunk 10]
```diff
@@ -0,0 +1,385 @@
...
+    Sys.iswindows() || (isinteractive() && MPI.finalize_atexit())
+    mpicomm=MPI.COMM_WORLD
+
+    ll = uppercase(get(ENV, "JULIA_LOG_LEVEL", "INFO"))
+    loglevel = ll == "DEBUG" ? Logging.Debug :
+    ll == "WARN"  ? Logging.Warn  :
+    ll == "ERROR" ? Logging.Error : Logging.Info
+    logger_stream = MPI.Comm_rank(mpicomm) == 0 ? stderr : devnull
+    global_logger(ConsoleLogger(logger_stream, loglevel))
+    @static if Base.find_package("CUDAnative") !== nothing
```
Comment (jkozdon) [inline test/DGmethods/sphere/advection_sphere_ssp34.jl]: if #246 lands first this should be
```suggestion
@static if haspkg("CUDAnative")
 ```

**Comment by jkozdon** [comment]
@fxgiraldo  I'm happy to make all these changes if you want me to.

File: test/DGmethods/runtests.jl (hunk 1 above)
Comment (fxgiraldo) [reply test/DGmethods/runtests.jl]: done.

File: test/DGmethods/runtests.jl (hunk 2 above)
Comment (fxgiraldo) [reply test/DGmethods/runtests.jl]: done

File: test/DGmethods/sphere/advection_sphere_lsrk.jl (hunk 5 above)
Comment (fxgiraldo) [reply test/DGmethods/sphere/advection_sphere_lsrk.jl]: I can hold off until #246 is accepted before doing this change so I won't have to modify stuff.

**Comment by jkozdon** [comment]
I'd probably do it with a 
```
git fetch origin
it rebase -i origin/master
```
Assuming origin is climate-machine. If not change appropriately. You could also squash all the commits into a single commit if you wanted (unless you want your commit history).

**Comment by fxgiraldo** [comment]
Fixed commit message to remove WIP.

File: test/DGmethods/sphere/advection_sphere_lsrk.jl [hunk 11]
```diff
@@ -0,0 +1,385 @@
...
+using LinearAlgebra
+using StaticArrays
+using Logging, Printf, Dates
+using Random
+
+@static if haspkg("CUDAnative")
+    using CUDAdrv
+    using CUDAnative
+    using CuArrays
+    const ArrayTypes = VERSION >= v"1.2-pre.25" ? (Array, CuArray) : (Array,)
```
Comment (simonbyrne) [inline test/DGmethods/sphere/advection_sphere_lsrk.jl]: I've changed the other examples to only use `CuArray`s on GPU nodes (to avoid double testing)

Comment (jkozdon) [reply test/DGmethods/sphere/advection_sphere_lsrk.jl]: OK. Want me to modify this PR and stop bors or do new PR?

Comment (simonbyrne) [reply test/DGmethods/sphere/advection_sphere_lsrk.jl]: Can be a new PR.

Comment (jkozdon) [reply test/DGmethods/sphere/advection_sphere_lsrk.jl]: Went ahead and fixed it since the CI timed out and I was sitting at the computer.

**Pull request merged**